src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private static final int RESIZE_VALUE = 2;
+    private Node<K, V>[] buckets;
+    private int size;
+
+    public MyHashMap() {
+        buckets = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
+        resizeIfNeeded();
+        int bucketIndex = calculateBucketIndex(key);
+        Node<K, V> existing = buckets[bucketIndex];
 
+        while (existing != null) {
+            if (isKeysEqual(existing.key, key)) {
+                existing.value = value;
+                return;
+            }
+            existing = existing.next;
+        }
+
+        buckets[bucketIndex] = new Node<>(key, value, buckets[bucketIndex]);
+        size++;
     }
 
     @Override
     public V getValue(K key) {
+        int bucketIndex = calculateBucketIndex(key);
+        Node<K, V> node = buckets[bucketIndex];
+
+        while (node != null) {
+            if (isKeysEqual(node.key, key)) {
```

Comment:
In this task you can use Objects.equals() as you wish

Reply:
done